### ISSUE TITLE
NixOS generates incorrect GRUB entries if / is on a btrfs subvolume

### ISSUE THREAD

**Issue by PkmX** [NixOS/nixpkgs#1696]
Hi,

I currently have a NixOS system where `/` is on a btrfs subvolume, and when it is booted with GRUB, GRUB [intentionally](http://savannah.gnu.org/bugs/?41449) always uses the real root of the btrfs filesystem. That means, assuming that my `/` is on a subvolume named `$subvol`, `/` of my NixOS system is actually `/$subvol` from GRUB's point of view. However, NixOS currently generates incorrect GRUB entries of the form:

```
linux /nix/store/<...>/bzImage systemConfig=/nix/store/<...> init=/nix/store/<...>/init loglevel=4
initrd /nix/store/<...>/initrd
```

, whereas it should actually be:

```
linux ${subvol}/nix/store/<...>/bzImage systemConfig=/nix/store/<...> init=/nix/store/<...>/init loglevel=4
initrd ${subvol}/nix/store/<...>/initrd
```

The included resources (fonts and background images) also need to have their paths modified so that they are relative to btrfs root as well.

**Comment by joachifm** [comment]
Is this still a problem?

**Comment by domenkozar** [comment]
I think this should be solved now.

**Comment by thanegill** [comment]
Can this be closed? Seems to be fixed by #1575

**Comment by thanegill** [comment]
Can this be closed? Seems to be fixed by #1575
